### PR TITLE
[Test] Remove JUnit

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

### Motivation

The codebase has two unit test frameworks: JUnit and TestNG, which always make us confused about which one to use. I check the git commits and found that #1032 does this thing, but still has some tests using JUnit. We need to remove the JUnit, use the TestNG as the unit test framework in the codebase.

### Modifications

- Remove JUnit dependency
- Clean the JUnit test in codebase

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Review by lhotari — CHANGES_REQUESTED** [review]
Please verify the correct parameter order for TestNG Asserts as explained in the comment above.

File: pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 import io.jsonwebtoken.SignatureAlgorithm;
 import org.apache.pulsar.broker.authentication.AuthenticationDataSource;
 import org.apache.pulsar.broker.authentication.utils.AuthTokenUtils;
-import org.junit.Assert;
```
[the remark this answers is not in the document]
Comment (Shoothzj) [reply pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java:25]: +1

[the remark this answers is not in the document]
Comment (nodece) [reply pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java:25]: Only `assertEquals` is affected, I updated all codebase.

**Review by Shoothzj — DISMISSED** [review]

**Review by nicoloboschi — APPROVED** [review]

File: pulsar-io/flume/src/test/java/org/apache/pulsar/io/flume/node/TestEnvVarResolverProperties.java [hunk 2]
```diff
@@ -18,32 +18,32 @@
...
-    public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+    public EnvironmentVariables environmentVariables;
     private PropertiesFileConfigurationProvider provider;
 
-    @Before
+    @BeforeMethod
     public void setUp() {
         provider = new PropertiesFileConfigurationProvider("a1", TESTFILE);
+        environmentVariables = new EnvironmentVariables();
+        System.out.println(environmentVariables);
```
Comment (nicoloboschi) [inline pulsar-io/flume/src/test/java/org/apache/pulsar/io/flume/node/TestEnvVarResolverProperties.java:39]: is it intentional ?

Comment (nodece) [reply pulsar-io/flume/src/test/java/org/apache/pulsar/io/flume/node/TestEnvVarResolverProperties.java]: Thanks, this has been removed.

**Review by nicoloboschi — DISMISSED** [review]

**Review by gaoran10 — APPROVED** [review]
LGTM

**Review by RobertIndie — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@lhotari Please help review again, thanks.

**Comment by codelipenghui** [comment]
@lhotari Please help review again.

File: pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClientSslTests.java [hunk 3]
```diff
@@ -18,6 +18,8 @@
  */
 package org.apache.pulsar.io.elasticsearch;
 
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertFalse;
```
Comment (nicoloboschi) [inline pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClientSslTests.java:22]:
```suggestion
import static org.testng.Assert.assertFalse;
```

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClientSslTests.java (hunk 3 above)
Comment (nodece) [reply pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchClientSslTests.java]: Thanks.

**Comment by codelipenghui** [comment]
@lhotari Please help review the PR again.

**Comment by nodece** [comment]
> Please verify the correct parameter order for TestNG Assert.assertEquals as explained in the comment above. Perhaps you already were aware of this since I see some changes in parameter order.

This request has been updated, could you continue to review?

**Comment by codelipenghui** [comment]
@lhotari @nicoloboschi Could you please help review the PR again?

**Review by nicoloboschi — APPROVED** [review]
nice! love to see dependencies dropped 😃

**Pull request closed without merging**